README.md

```diff
@@ -1,85 +1,167 @@
+(which is an object) on the `Order` class:
 
+```js
+class Order extends React.Component {
   render() {
-    // blah blah
+    // ...
   }
 }
+
+Order.defaultProps = {
+  cone: true,
+  size: 'regular'
+};
+
+Order.propTypes = {};
+```
+
+Let's add our first PropType, the `cone` one:
+
+```js
+Order.propTypes = {
+  cone: React.PropTypes.bool
+};
 ```
 
-But if we have a bigger `User` component that takes the whole `user` object:
+We just told our component to expect the `cone` prop to be a boolean. There are a couple of important things to note
+here:
+ 
+- Notice the capitalization: the property on the component class is always called `propTypes`, while
+the React prop types you assign to them are capitalized as `PropTypes`. This is easy to mess up, so if our propTypes
+don't validate, that's the first place we should look!
+- Some types are not called what you'd expect them to be: a boolean is `bool`, a function is `func`, and so on. Using
+the [reference on PropTypes][proptypes-reference] is
+probably a good idea if you're just starting out with this stuff.
+
+Let's add our second prop, the `size` one, which expects a string:
+
+```js
+Order.propTypes = {
+  cone: React.PropTypes.bool,
+  size: React.PropTypes.string
+};
+```
 
-```javascript
-const React = require('react')
-const PropTypes = React.PropTypes
+Next up is our scoops array:
 
-const Address = require('./components/Address.jsx')
+```js
+Order.propTypes = {
+  cone: React.PropTypes.bool,
+  size: React.PropTypes.string,
+  scoops: React.PropTypes.array.isRequired
+};
+```
 
-class User extends React.Component {
-  // as with defaultProps, we'll need to include
-  // the correct transform for this to work
-  propTypes = {
-    user: PropTypes.shape({
-      address: PropTypes.object.isRequired
-    })
-  }
+See the `isRequired` there? That makes sure the `scoops` prop is provided when rendering the component — we can't have
+an order with no scoops of ice cream! To further validate the `scoops` prop, let's tell it we're expecting an _array of
+strings_:
 
-  render() {
-    return <Address address={this.props.user.address} />
-  }
-}
+```js
+Order.propTypes = {
+  cone: React.PropTypes.bool,
+  size: React.PropTypes.string,
+  scoops: React.PropTypes.arrayOf(React.PropTypes.string).isRequired
+};
+```
+
+That's better! Next, let's finish up by adding the last prop, the `orderInfo`:
+
+```js
+Order.propTypes = {
+  cone: React.PropTypes.bool,
+  size: React.PropTypes.string,
+  scoops: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
+  orderInfo: React.PropTypes.object.isRequired
+};
 ```
 
-We only need to validate that `user.address` is defined here; we can wait until
-`Address` renders to handle the details.
+We've now fully validated all of the props that our `<Order />` component cares about. Now we can be sure that the right
+data gets passed in, and if not, we can change things accordingly!
+
+## Defining shape for object PropTypes
```

Comment:
Minor minor suggestion: maybe put scare quotes around shape, since it's a specific and somewhat idiosyncratic term...